### PR TITLE
Cleanup runtime platform classes

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

1. Merges ```PlatformSupport``` project into the ```Avalonia.Base```. ```AppBuilder``` was moved to the ```Avalonia.Controls```. PlatformSupport was previously added temporary to simplify new android backends and before core libraries were merged.
2. Where possible NET6/NETSTANDARD implementations of runtime platform interfaces were used. To be specific:
- Uses [NativeLibrary](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativelibrary?view=net-6.0) to load library or export proc. **Only NET6** (technically netcore3.0+, but still 'no' for us).
- [RuntimeFeature.IsDynamicCodeSupported](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodesupported?view=net-6.0) to know if it's possible to load assembly or not. **Only NET6** (was added with netstandard2.1).
- ~Marshal.AllocHGlobal instead of mmap on linux in UnmanagedBlob.~ Reverted after a discussion.
3. Removes legacy xamarin solution configuration (iPhoneSimulator, Ad-Hoc...). They are not used with new mobile backend, and we don't need them anyway.
4. Deprecated overload of AppBuilderBase.Start was removed (deprecated since 0.9.0)

## Breaking changes

Only binary and only in nightly builds. So, no.

### CODE REVIEW

File: src/Avalonia.Base/Platform/Internal/DynLoader.cs [hunk 1]
```diff
@@ -26,25 +26,6 @@ public static void Init()
             }
         }
 
-        static class AndroidImports
```
Comment (maxkatz6) [inline src/Avalonia.Base/Platform/Internal/DynLoader.cs:29]: NativeLibrary API is always available for net6-android, so it is used instead.

File: src/Avalonia.Base/Platform/StandardRuntimePlatform.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        public IUnmanagedBlob AllocBlob(int size) => new UnmanagedBlob(size);
+        
+        private static readonly Lazy<RuntimePlatformInfo> Info = new(() =>
+        {
+            OperatingSystemType os;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                os = OperatingSystemType.OSX;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")))
```
Comment (kekekeks) [inline src/Avalonia.Base/Platform/StandardRuntimePlatform.cs:23]: I don't think FreeBSD counts as Linux

File: src/Avalonia.Base/Platform/StandardRuntimePlatform.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("iOS")))
+                os = OperatingSystemType.iOS;
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Create("Browser")))
+                os = OperatingSystemType.Browser;
+            else
+                throw new Exception("Unknown OS platform " + RuntimeInformation.OSDescription);
+
+            // Source: https://github.com/dotnet/runtime/blob/main/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs
+            var isCoreClr = Environment.Version.Major >= 5 || RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.OrdinalIgnoreCase);
+            var isMonoRuntime = Type.GetType("Mono.RuntimeStructs") != null;
```
Comment (kekekeks) [inline src/Avalonia.Base/Platform/StandardRuntimePlatform.cs:38]: I think the traditional way is to check for `Mono.Runtime` - https://www.mono-project.com/docs/gui/winforms/porting-winforms-applications/#runtime-conditionals

File: src/Avalonia.Base/Platform/StandardRuntimePlatform.cs (hunk 2 above)
Comment (maxkatz6) [reply src/Avalonia.Base/Platform/StandardRuntimePlatform.cs:23]: Seems like a mistake from previous PR. Will remove from here.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020671-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020694-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020754-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0021076-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Pull request merged**